cishouseholds/pipeline/weight_calibration.py

```diff
@@ -0,0 +1,104 @@
+import rpy2.robjects as robjects
+from rpy2.robjects.conversion import localconverter
+from rpy2.robjects.packages import importr
+
+from cishouseholds.pipeline.load import extract_from_table
+from cishouseholds.pipeline.load import update_table
+from cishouseholds.pipeline.pipeline_stages import register_pipeline_stage
+from cishouseholds.pyspark_utils import get_or_create_spark_session
+
+regenesees = importr(
+    "ReGenesees",
+    # Fix conflicting method names in R
+    robject_translations={
+        "dim.GVF.db": "dim_GVF_db_",
+        "print.GVF.db": "print_GVF_db_",
+        "str.GVF.db": "str_GVF_db_",
+    },
+)
+
+
+@register_pipeline_stage("weight_calibration")
+def weight_calibration(calibration_config: dict):
+    """
+    Run weight calibration for multiple datasets, as specified by the stage configuration.
+    """
+    spark_session = get_or_create_spark_session()
+    population_totals_df = extract_from_table(calibration_config["population_totals_table_name"]).toPandas()
+    with localconverter(robjects.default_converter + robjects.pandas2ri.converter):
+        population_totals_df = robjects.conversion.py2rpy(population_totals_df)
+
+    for dataset_options in calibration_config["dataset_options"]:
+        pandas_df = extract_from_table(dataset_options["input_table_name"]).toPandas()
```

Comment:
Move extract above, then subset within loop. Or extract subset directly based on subset column flag (1)

Also needs to read in population totals and pivot them